### ISSUE TITLE
geodreckon - bug when azimuth is 0 and optional output S12 is requested

### ISSUE THREAD

**Issue by xpyctjr** [geographiclib/geographiclib-octave#1]
I've found a bug in `geodreckon` when the optional output `S12` is requested and the `azi1` input is, or is an array with a 0.

The following code should replicate the issue.

    lat = -33.911143;
    lon = 151.225339;
    distances = linspace(0,30000*1000,100);
    azi = [0 1 2 3 4]; % Some azimuth array with a 0
    [lats,lons, azi2, S12, m12, M12, M21, a12_s12] = geodreckon(lat,lon,distances',[0 1]);

Running the above code should yield the following error.

``` 
The logical indices contain a true value outside of the array bounds.

Error in geodreckon (line 301)
    salp12(s) = salp2(s) .* calp1(s) - calp2(s) .* salp1(s);
```

Let me know if you need anything else! I really like the library - it has been super useful.
This is my first bug/issue report, so let me know if I've missed anything or committed a social faux paus.

**Comment by cffk** [comment]
Thanks for finding this bug and for your detailed bug report.  I've fixed the bug and will push out a new version of the package soon.  If you want to try the fix, apply this patch:

```patch
diff --git a/inst/geodreckon.m b/inst/geodreckon.m
index 49a6010..37256f9 100644
--- a/inst/geodreckon.m
+++ b/inst/geodreckon.m
@@ -298,6 +298,7 @@ function [lat2, lon2, azi2, S12, m12, M12, M21, a12_s12] = geodreckon ...
                    csig12 <= 0);
     calp12 = salp0.^2 + calp0.^2 .* csig1 .* csig2;
     s = calp0 == 0 | salp0 == 0;
+    salp1 = expand(salp1, S1, Z); calp1 = expand(calp1, S1, Z);
     salp12(s) = salp2(s) .* calp1(s) - calp2(s) .* salp1(s);
     calp12(s) = calp2(s) .* calp1(s) + salp2(s) .* salp1(s);
     if e2 ~= 0
```

P.S. It's "faux pas" (meaning "false step" in French), not "faux paus".

**Comment by cffk** [comment]
The bug fix is in Version 2.3.2 released on 2024-10-05.
